### ISSUE TITLE
Unschedule action only by hook name

### ISSUE THREAD

**Issue by crstauf** [woocommerce/action-scheduler#417]
The [documentation for `as_unschedule_action()`](https://github.com/woocommerce/action-scheduler/blob/master/docs/api.md#function-reference--as_unschedule_action) indicates that only the hook name is required, but in my quick test, the other arguments are needed as well.

Is the documentation incorrect, or should actions be able to be unscheduled with only the hook name?

**Comment by rrennick** [comment]
@crstauf Thanks for reporting this. I was able to reproduce this with

```
add_action( 'init', function() {
	as_schedule_single_action( time(), 'unschedule_test_action' );
	$ids = as_get_scheduled_actions( [ 'hook' => 'unschedule_test_action' ], 'ids' );
	error_log( count( $ids ) );
	as_unschedule_action( 'unschedule_test_action' );
	$ids = as_get_scheduled_actions( [ 'hook' => 'unschedule_test_action' ], 'ids' );
	error_log( count( $ids ) );
});
```

Log entries:

```
[24-Jan-2020 15:16:16 UTC] 1
[24-Jan-2020 15:16:16 UTC] 1
```

**Comment by crstauf** [comment]
> Is the documentation incorrect, or should actions be able to be unscheduled with only the hook name?

Actions should be able to be unscheduled using only the action name (based on [above response](https://github.com/woocommerce/action-scheduler/issues/417#issuecomment-578172625) and adding "bug" label).

**Comment by crstauf** [comment]
@rrennick Started working on a fix for the bug, and realized the above reproduction code is faulty: `as_get_scheduled_actions()` returns actions regardless of status, so the count includes cancelled actions

### Corrected Version (of [original](https://github.com/woocommerce/action-scheduler/issues/417#issuecomment-578172625))
```php
add_action( 'init', function() {
	as_schedule_single_action( time() + 10, 'unschedule_test_action' );
	$ids = as_get_scheduled_actions( [ 'hook' => 'unschedule_test_action' ], 'ids' );
	error_log( count( $ids ) );
	as_unschedule_action( 'unschedule_test_action' );
	$ids = as_get_scheduled_actions( [ 'hook' => 'unschedule_test_action', 'status' => ActionScheduler_Store::STATUS_PENDING ], 'ids' );
	error_log( count( $ids ) );
} );
```

Log entries:

```
[28-Feb-2020 03:18:44 UTC] 1
[28-Feb-2020 03:18:44 UTC] 0
```

To reproduce the bug, the `as_scheduled_single_action()` call should include arguments, and the `as_unschedule_action()` call should not.

### Reproduce Bug
```php
add_action( 'init', function() {
	as_schedule_single_action( time() + 10, 'unschedule_test_action', array( 'foo' ) );
	$ids = as_get_scheduled_actions( [ 'hook' => 'unschedule_test_action' ], 'ids' );
	error_log( count( $ids ) );
	as_unschedule_action( 'unschedule_test_action' );
	$ids = as_get_scheduled_actions( [ 'hook' => 'unschedule_test_action', 'status' => ActionScheduler_Store::STATUS_PENDING ], 'ids' );
	error_log( count( $ids ) );
} );
```

Log entries:

```
[28-Feb-2020 03:20:22 UTC] 1
[28-Feb-2020 03:20:22 UTC] 1
```

Continuing to investigate a resolution for the bug.

**Comment by crstauf** [comment]
Submitted and linked [PR](https://github.com/woocommerce/action-scheduler/pull/475).

**Comment by leewillis77** [comment]
Can't you just change the `is_array( $args )` to `!empty( $args )` so it works the same as `as_unschedule_all_actions` which is unaffected by this issue? https://github.com/woocommerce/action-scheduler/blob/master/functions.php#L115

**Comment by rrennick** [comment]
Addressed in #475
